### PR TITLE
Don't wait for analytics submission

### PR DESCRIPTION
**Pull request by nirinchev**
## Description

Fixes #

##  TODO

* [ ] Changelog entry
* [ ] Tests (if applicable)

### CODE REVIEW

File: Realm/Realm.Weaver/Analytics/Analytics.cs [hunk 1]
```diff
@@ -80,18 +80,24 @@ internal class Analytics
...
+
+            _references = references;
             _logger = logger;
 
             _realmFeaturesToAnalyze = Metric.SdkFeatures.Keys.ToDictionary(c => c, _ => (byte)0);
 
             _classAnalysisSetters = new()
             {
-                [Feature.IEmbeddedObject] = member => (true, Feature.IEmbeddedObject),
-                [Feature.IAsymmetricObject] = member => (true, Feature.IAsymmetricObject),
```
Comment (LaPeste) [inline Realm/Realm.Weaver/Analytics/Analytics.cs:94]: Good catch

File: Realm/Realm.Weaver/Analytics/Analytics.cs [hunk 2]
```diff
@@ -239,7 +250,7 @@ internal void AnalyzeUserAssembly(ModuleDefinition module)
                 _realmEnvMetrics[UserEnvironment.NetFramework] = _config.TargetFramework;
                 _realmEnvMetrics[UserEnvironment.NetFrameworkVersion] = _config.TargetFrameworkVersion;
 
-                foreach (var type in module.Types)
+                foreach (var type in module.Types.ToArray())
```
Comment (LaPeste) [inline Realm/Realm.Weaver/Analytics/Analytics.cs:253]: Why is this necessary?

**Review by LaPeste — COMMENTED** [review]
The main idea seems good. We need to investigate how fast it's now VS before. Mostly curiosity as this improvement was almost for free.
I'd be interesting to investigate what happens in a console build, as you mentioned.

File: Realm/Realm.Weaver/Analytics/Analytics.cs (hunk 2 above)
Comment (nirinchev) [reply Realm/Realm.Weaver/Analytics/Analytics.cs:253]: This is executing while the weaver is running, which means that if the weaver adds new types (and it does for the schema), iteration will break.

Comment (LaPeste) [reply Realm/Realm.Weaver/Analytics/Analytics.cs:253]: Makes sense. Thank you.
